### PR TITLE
Add basic linear LoRA support for Stable Diffusion v1 UNet

### PR DESCRIPTION
**Pull request by RyanJDick**
- Add ability to inject LoRA layers into a Stable Diffusion v1 UNet.
- Add converter to kohya_ss state_dict format for SD v1 UNet LoRA layers.

### CODE REVIEW

File: src/invoke_training/lora/injection/lora_layer_collection.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class LoRALayerCollection(torch.nn.Module):
+    """A collection of LoRA layers (with names). Typically used to perform operations on a group of LoRA layers during
+    training.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        # A torch.nn.ModuleDict may seem like a more natural choice here, but it does not allow keys that contain '.'
```
Comment (brandonrising) [inline src/invoke_training/lora/injection/lora_layer_collection.py:16]: Would a normal `dict[str, torch.nn.ModuleList]` work? I just think depending on the order of two arrays matching could lead to some weird edge cases.

**Review by brandonrising — APPROVED** [review]
lgtm

File: src/invoke_training/lora/injection/lora_layer_collection.py (hunk 1 above)
Comment (RyanJDick) [reply src/invoke_training/lora/injection/lora_layer_collection.py:16]: Unfortunately not, module in a dict would not be properly registered with the parent torch.nn.Module, so methods like `.train()`, `.parameters()`, etc. wouldn't work. I've updated the comment to make this more clear in [1cee5a7](https://github.com/invoke-ai/invoke-training/pull/5/commits/1cee5a7cdbd036576943808b7c028b50f4c3c288)

File: tests/invoke_training/lora/injection/test_stable_diffusion_v1.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    # assert len(lora_layers) == 192 # TODO(ryand): Enable this check once conv layers are added.
+    for layer_name in lora_layers._names:
+        assert layer_name.endswith(("to_q", "to_k", "to_v", "to_out.0", "ff.net.0.proj", "ff.net.2"))
+
+
+@pytest.mark.loads_model
+def test_convert_lora_state_dict_to_kohya_format_sd1_smoke():
+    """Smoke test of convert_lora_state_dict_to_kohya_format_sd1(...) with full SD 1.5 model."""
+    unet = UNet2DConditionModel.from_pretrained(
+        "runwayml/stable-diffusion-v1-5", subfolder="unet", local_files_only=True
```
[the remark this answers is not in the document]
Comment (RyanJDick) [reply tests/invoke_training/lora/injection/test_stable_diffusion_v1.py]: Good call, added in [5fdc63b](https://github.com/invoke-ai/invoke-training/pull/5/commits/5fdc63b3c964e676ea459efa4854252fa0157e41)

**Pull request merged**